pkg/cloudprovider/types.go

```diff
@@ -231,7 +231,7 @@ type Offerings []Offering
 // passed zone and capacity type
 func (ofs Offerings) Get(ct, zone string) (Offering, bool) {
```

Comment:
I know we had discussed leaving this function as is previously, but having thought about it some more I think we probably should make this generic across offering constraints for consistency. I think this would be simple enough to handle by constructing a set of requirements from the disruption candidate labels and passing them rather than the capacity type and zone.

Reply:
I think we should consider leaving Zone and CapacityType as the options in Get() and Compatible(). More reasoning above.

Reply:
Discussed offline - generalized in next commit.